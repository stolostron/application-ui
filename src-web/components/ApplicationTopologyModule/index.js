/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/

export {
  default as ApplicationTopologyModule
} from './ApplicationTopologyModule'
