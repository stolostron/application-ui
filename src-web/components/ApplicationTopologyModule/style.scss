/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
@import 'node_modules/carbon-components/scss/globals/scss/_colors';
@import 'scss/variables';

.resourceDiagramSourceContainer {
  position: relative;

  /* diagram and controls */
  .resourceDiagramControlsContainer {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    position: relative;
    box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.10);

    /* title */
    .diagram-title {
      background-color: #fff;
      font-weight: bold;
      font-size: 14px;
      border-bottom: 1px solid #dfe3e6;
      padding: 15px;
    }

    .diagram-title-divider {
      background-color: #dcdcdc;
      width: 1px;
      height: 17px;
      margin-left: 20px;
      margin-right: 20px;
      vertical-align: text-bottom;
    }

    .how-to-read-text {
      color: #3d70b2;
      font-size: 14px;
      line-height: 20px;
      text-align: left;
      cursor: pointer;
      margin-right: 4px;
    }

    .how-to-read-icon {
      width: 14px;
      height: 11px;
      margin-left: 9px;
    }

    .bx--tooltip__trigger svg {
      height: 14px;
    }

    /* filter bar */
    .diagram-controls-container {
      background-color: #fff;
      display: flex;
      align-items: center;
      justify-content: space-between;
      padding: 10px 20px 10px 10px;
      border-top: 1px solid #dfe3e6;

      .diagram-type-filter-bar {
        display: flex;
      }

      .diagram-expand-button {
        display: flex;
        font-size: 12px;
        cursor: pointer;
        color: #3d70b2;

        .icon {
          margin-right: 8px;
          width: 13px;
          height: 13px;
          stroke: #3d70b2;
        }
      }
    }
  }

  /* source editor */
  .resourceEditorContainer {
    width: 100%;
    height: calc(100vh - 200px);
    display: flex;
    flex-direction: column;
    position: relative;

    .resource-editor-toolbar {
      display: flex;
      align-items: center;
      margin-top: 26px;
    }

    .bx--inline-notification {
      position: absolute;
      top: 16px;
      left: 16px;
      z-index: 1000;
      background-color: white;
    }
  }

  /* when editor is open */
  &.showExpandedTopology {
    height: calc(100vh - 213px);
    margin: -30px -4rem -70px -3rem;

    /* when editor is open, diagram and control positions */
    .resourceDiagramControlsContainer {
      height: calc(100vh - 225px);
      box-shadow: none;

      /* title */
      .diagram-title {
        display: none;
      }

      .topology-container {
        box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.10);
      }


      /* filter bar */
      .diagram-controls-container {
        border: none;
        background-color: unset;
        padding: 0 0 6px;
      }
    }
  }

  .last-update-container {
    position: absolute;
    right: 20px;
    top: 10px;
    z-index: 1000;
    display: flex;
    align-items: center;
    font-size: 9px;
    line-height: 12px;

    .bx--loading--small {
      width: 1rem;
      height: 1rem;
      margin: -2px 0;
    }
  }

  /* when view is split, show details view on other side of diagram */
  &.split {
    .topologyDetails {
      left: -200px;
      right: unset;
    }
  }

  .yamlEditorContainer {
    height: calc(100% - 36px);

    & * {
      font-family: unset;
    }
  }

  .Resizer {
    width: 0;
    opacity: .2;
    z-index: 1;
    -moz-box-sizing: border-box;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    -moz-background-clip: padding;
    -webkit-background-clip: padding;
    background-clip: padding-box;
    border-left: 3px solid rgba(255, 255, 255, 0);
    border-right: 3px solid rgba(255, 255, 255, 0);
    cursor: col-resize;
    margin: 0 5px;
  }

  .Resizer:hover {
    -webkit-transition: all .2s ease;
    transition: all .2s ease;
  }

  .Resizer.vertical:hover {
    background: rgba(0, 0, 0, 0.5);
  }
}
