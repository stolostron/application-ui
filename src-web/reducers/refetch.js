/*******************************************************************************
 * Copyright (c) 2020 Red Hat, Inc.
 *******************************************************************************/

import { REFETCH_INTERVAL_UPDATE } from '../actions'

export const refetch = (state, action) => {
  switch (action.type) {
  case REFETCH_INTERVAL_UPDATE:
    return action.data
  default:
    // get the poll interval from RefreshTimeSelect component
    return Object.assign({}, state, {})
  }
}
