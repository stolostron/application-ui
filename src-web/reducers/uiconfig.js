/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/

import { UICONFIG_RECEIVE_SUCCESS } from '../actions'

export const uiconfig = (state = null, action) => {
  switch (action.type) {
  case UICONFIG_RECEIVE_SUCCESS:
    return action.data
  default:
    return state
  }
}
