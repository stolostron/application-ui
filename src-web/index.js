/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
import React from 'react'
import { hydrate } from 'react-dom'
import { Provider } from 'react-redux'
import thunkMiddleware from 'redux-thunk'
import { createLogger } from 'redux-logger'
import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import { Router } from 'react-router-dom'
import { ApolloProvider } from 'react-apollo'
import App from './containers/App'
import ScrollToTop from './components/common/ScrollToTop'
import * as reducers from './reducers'
import config from '../lib/shared/config'
import apolloClient from '../lib/client/apollo-client'
import history from 'history'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

window.SHARED_HISTORY = window.SHARED_HISTORY
  ? window.SHARED_HISTORY
  : history.createBrowserHistory()

const loggerMiddleware = createLogger()
// Grab the state from a global variable injected into the server-generated HTML
const preloadedState = window.__PRELOADED_STATE__

const middleware = [thunkMiddleware] // lets us dispatch() functions
if (
  !window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
  config['featureFlags:reduxLogger']
) {
  middleware.push(loggerMiddleware) // middleware that logs actions
}

// Allow the passed state to be garbage-collected
delete window.__PRELOADED_STATE__

// Create Redux store with initial state
const store = createStore(
  combineReducers(reducers),
  preloadedState,
  composeEnhancers(applyMiddleware(...middleware))
)

hydrate(
  <ApolloProvider client={apolloClient.getClient()}>
    <Provider store={store}>
      <Router history={window.SHARED_HISTORY}>
        <ScrollToTop>
          <App />
        </ScrollToTop>
      </Router>
    </Provider>
  </ApolloProvider>,
  document.getElementById('page')
)
