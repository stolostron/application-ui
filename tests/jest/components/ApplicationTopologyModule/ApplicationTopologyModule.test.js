/*******************************************************************************
 * Licensed Materials - Property of Red Hat
 * Copyright (c) 2020 Red Hat, Inc.
 *******************************************************************************/
jest.mock("../../../../lib/client/apollo-client", () => ({
  getSearchClient: jest.fn(() => {
    return null;
  }),
  getResource: jest.fn((resourceType, { namespace }) => {
    if (
      resourceType === "channel" ||
      (resourceType.name && resourceType.name === "HCMChannel")
    ) {
      const channelData = {
        data: {
          searchResult: [
            {
              items: [
                {
                  kind: "channel",
                  name: "mortgage-channel",
                  namespace: "mortgage-ch",
                  _hubClusterResource: "true"
                }
              ]
            }
          ]
        }
      };
      return Promise.resolve(channelData);
    }

    if (
      resourceType === "subscription" ||
      (resourceType.name && resourceType.name === "HCMSubscription")
    ) {
      const subscriptionData = {
        data: {
          searchResult: [
            {
              items: [
                {
                  kind: "subscription",
                  name: "orphan",
                  namespace: "default",
                  status: "Propagated",
                  cluster: "local-cluster",
                  channel: "default/mortgage-channel",
                  apigroup: "app.ibm.com",
                  apiversion: "v1alpha1",
                  _rbac: "default_app.ibm.com_subscriptions",
                  _hubClusterResource: "true",
                  _uid:
                    "local-cluster/5cdc0d8d-52aa-11ea-bf05-00000a102d26orphan",
                  packageFilterVersion: ">=1.x",
                  label:
                    "app=mortgage-app-mortgage; chart=mortgage-1.0.3; heritage=Tiller; release=mortgage-app",
                  related: []
                }
              ]
            }
          ]
        }
      };
      return Promise.resolve(subscriptionData);
    }

    if (
      resourceType === "placementrule" ||
      (resourceType.name && resourceType.name === "HCMPlacementRule")
    ) {
      const prData = {
        data: {
          searchResult: [
            {
              items: [
                {
                  kind: "subscription",
                  name: "orphan",
                  namespace: "default",
                  status: "Propagated",
                  cluster: "local-cluster",
                  channel: "default/mortgage-channel",
                  apigroup: "app.ibm.com",
                  apiversion: "v1alpha1",
                  _rbac: "default_app.ibm.com_subscriptions",
                  _hubClusterResource: "true",
                  _uid:
                    "local-cluster/5cdc0d8d-52aa-11ea-bf05-00000a102d26orphan",
                  packageFilterVersion: ">=1.x",
                  label:
                    "app=mortgage-app-mortgage; chart=mortgage-1.0.3; heritage=Tiller; release=mortgage-app",
                  related: []
                }
              ]
            }
          ]
        }
      };
      return Promise.resolve(prData);
    }

    return Promise.resolve({ response: "invalid resonse" });
  })
}));

const React = require("../../../../node_modules/react");

import ApplicationTopologyModule from "../../../../src-web/components/ApplicationTopologyModule/ApplicationTopologyModule.js";

import renderer from "react-test-renderer";
import * as reducers from "../../../../src-web/reducers";
import { mount } from "enzyme";

import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import thunkMiddleware from "redux-thunk";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

const preloadedState = window.__PRELOADED_STATE__;
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const middleware = [thunkMiddleware];

const store = createStore(
  combineReducers(reducers),
  preloadedState,
  composeEnhancers(applyMiddleware(...middleware))
);

const locale = "en-US";
describe("ApplicationTopologyModule with selected node ID", () => {
  it("ApplicationTopologyModule renders correctly when topology is not expanded", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <ApplicationTopologyModule
              selectedNodeId={nodeID}
              showExpandedTopology={false}
              params={params}
              locale={locale}
            />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("ApplicationTopologyModule renders correctly when topology is expanded", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <ApplicationTopologyModule
              selectedNodeId={nodeID}
              showExpandedTopology={true}
              params={params}
              locale={locale}
            />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  const actions = {
    setShowExpandedTopology: jest.fn()
  };
  it("ApplicationTopologyModule renders correctly when topology is expanded click", () => {
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <ApplicationTopologyModule
            selectedNodeId={nodeID}
            showExpandedTopology={true}
            params={params}
            locale={locale}
            actions={actions}
          />
        </Provider>
      </BrowserRouter>
    );

    wrapper
      .find(".bx--search-input")
      .at(0)
      .simulate("change");
    wrapper
      .find(".bx--search-close")
      .at(0)
      .simulate("click");

    wrapper
      .find(".diagram-collapse-button")
      .at(0)
      .simulate("click");

    wrapper
      .find(".diagram-collapse-button")
      .at(0)
      .simulate("keypress");
  });
});

const nodeID = "acb123";

const params = {
  name: "app1",
  namespace: "default"
};
