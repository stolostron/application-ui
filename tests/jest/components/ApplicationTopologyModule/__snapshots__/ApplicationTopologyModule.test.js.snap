// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationTopologyModule with selected node ID ApplicationTopologyModule renders correctly when topology is expanded---bbb 1`] = `
<div
  className="resourceDiagramSourceContainer showExpandedTopology split"
>
  <div
    className="SplitPane  vertical "
    style={
      Object {
        "MozUserSelect": "text",
        "WebkitUserSelect": "text",
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "left": 0,
        "msUserSelect": "text",
        "outline": "none",
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "userSelect": "text",
      }
    }
  >
    <div
      className="Pane vertical Pane1  "
      style={
        Object {
          "flex": "none",
          "outline": "none",
          "position": "relative",
          "width": 1000,
        }
      }
    >
      <div
        className="resourceDiagramControlsContainer"
      >
        <div
          className="diagram-title"
        >
          Resource topology
        </div>
        <div
          className="diagram-controls-container"
        >
          <div
            className="diagram-type-filter-bar"
            id="type-filter-bar-portal-id"
          />
        </div>
        <div
          className="topology-container"
        >
          <div
            className="topologyDiagramContainer"
          >
            <div
              className="diagramViewerDiagram"
            >
              <div
                className="diagramViewerContainerContainer"
                id="diagram-viewer-container-container"
              >
                <div
                  aria-label="zoom"
                  className="diagramViewerContainer"
                  role="region"
                  style={
                    Object {
                      "height": "100%",
                      "width": "100%",
                    }
                  }
                >
                  <svg
                    className="topologyDiagram"
                    id="topologySvgId"
                  />
                </div>
                <div
                  className="secondaryLoad"
                >
                  <div
                    aria-live="assertive"
                    className="bx--loading"
                  >
                    <svg
                      className="bx--loading__svg"
                      viewBox="-75 -75 150 150"
                    >
                      <circle
                        cx="0"
                        cy="0"
                        r="37.5"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  className="layoutLoadingContainer"
                >
                  <div
                    aria-live="assertive"
                    className="bx--loading"
                  >
                    <svg
                      className="bx--loading__svg"
                      viewBox="-75 -75 150 150"
                    >
                      <circle
                        cx="0"
                        cy="0"
                        r="37.5"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                className="diagram-controls"
              >
                <div
                  aria-label="Zoom in"
                  className="zoom-in-button rectangle-zoom"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                  title="Zoom in"
                >
                  <svg
                    className="icon"
                  >
                    <use
                      href="#diagramIcons_zoomIn"
                    />
                  </svg>
                </div>
                <div
                  aria-label="Zoom out"
                  className="zoom-out-button rectangle-zoom"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                  title="Zoom out"
                >
                  <svg
                    className="icon"
                  >
                    <use
                      href="#diagramIcons_zoomOut"
                    />
                  </svg>
                </div>
                <div
                  aria-label="Zoom to fit"
                  className="zoom-target-button rectangle-zoom"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                  title="Zoom to fit"
                >
                  <svg
                    className="icon_target"
                  >
                    <use
                      href="#diagramIcons_zoomTarget"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      className="Resizer vertical "
      onClick={[Function]}
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    />
    <div
      className="Pane vertical Pane2  "
      style={
        Object {
          "flex": 1,
          "outline": "none",
          "position": "relative",
        }
      }
    >
      <div
        className="resourceEditorContainer"
      >
        <div
          className="creation-view-yaml-header"
        >
          <div
            className="creation-view-yaml-header-toolbar"
            role="region"
          >
            <div
              className="editor-bar"
            >
              <div
                className="editor-bar-title"
              >
                YAML
              </div>
              <div
                className="editor-bar-toolbar"
              >
                <div
                  className="editor-bar-section"
                >
                  <div
                    aria-label="Reset"
                    className="editor-bar-button  disabled"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    role="button"
                    tabIndex={0}
                    title="Reset"
                  >
                    <div>
                      Reset
                    </div>
                  </div>
                </div>
                <div
                  className="editor-bar-section"
                >
                  <div
                    aria-label="Undo"
                    className="editor-bar-button  disabled"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    role="button"
                    tabIndex={0}
                    title="Undo"
                  >
                    <svg>
                      <use
                        href="#diagramIcons_undo"
                      />
                    </svg>
                  </div>
                  <div
                    aria-label="Redo"
                    className="editor-bar-button  disabled"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    role="button"
                    tabIndex={0}
                    title="Redo"
                  >
                    <svg>
                      <use
                        href="#diagramIcons_redo"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  className="editor-bar-section"
                >
                  <div
                    aria-label="Find"
                    className="editor-bar-search"
                    id="Find"
                    role="region"
                  >
                    <div
                      aria-labelledby="template-editor-search-undefined-label"
                      className="bx--search bx--search--sm"
                      role="search"
                    >
                      <svg
                        alt=""
                        aria-label=""
                        className="bx--search-magnifier"
                        fillRule="evenodd"
                        height="16"
                        role="img"
                        viewBox="0 0 16 16"
                        width="16"
                      >
                        <title>
                          
                        </title>
                        <path
                          d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zm4.936-1.27l4.563 4.557-.707.708-4.563-4.558a6.5 6.5 0 1 1 .707-.707z"
                        />
                      </svg>
                      <label
                        className="bx--label"
                        htmlFor="template-editor-search-undefined"
                        id="template-editor-search-undefined-label"
                      >
                        
                      </label>
                      <input
                        aria-label="Find"
                        className="bx--search-input"
                        id="template-editor-search-undefined"
                        onChange={[Function]}
                        placeholder="Find"
                        spellCheck={false}
                        type="text"
                      />
                      <button
                        className="bx--search-close bx--search-close--hidden"
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          alt="Provide a description that will be used as the title"
                          aria-label="Provide a description that will be used as the title"
                          fillRule="evenodd"
                          height="16"
                          role="img"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <title>
                            Provide a description that will be used as the title
                          </title>
                          <path
                            d="M8 6.586L5.879 4.464 4.464 5.88 6.586 8l-2.122 2.121 1.415 1.415L8 9.414l2.121 2.122 1.415-1.415L9.414 8l2.122-2.121-1.415-1.415L8 6.586zM8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"
                          />
                        </svg>
                      </button>
                    </div>
                    <div
                      aria-label="Previous"
                      className="editor-bar-button  disabled"
                      onClick={[Function]}
                      onKeyPress={[Function]}
                      role="button"
                      tabIndex={0}
                      title="Previous"
                    >
                      <svg>
                        <use
                          href="#diagramIcons_previous"
                        />
                      </svg>
                    </div>
                    <div
                      aria-label="Next"
                      className="editor-bar-button  disabled"
                      onClick={[Function]}
                      onKeyPress={[Function]}
                      role="button"
                      tabIndex={0}
                      title="Next"
                    >
                      <svg>
                        <use
                          href="#diagramIcons_next"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  className="editor-bar-section diagram-close-button"
                >
                  <div
                    className="editor-bar-button "
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <svg>
                      <use
                        href="#diagramIcons_close"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        
        <div
          className="yamlEditorContainer"
        >
          <div
            className="react-monaco-editor-container"
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ApplicationTopologyModule with selected node ID ApplicationTopologyModule renders correctly when topology is not expanded---aaa 1`] = `
<div
  className="resourceDiagramSourceContainer"
>
  <div
    className="topology-controls"
  >
    <div
      className="topology-control-container"
    >
      <div
        id="search-textbox-portal-id"
      />
    </div>
  </div>
  <div
    className="resource-toolbar"
    id="resource-toolbar"
  >
    <div
      className="resource-toolbar-container"
    >
      <div
        className="resource-toolbar-buttons"
      >
        <div
          id="assorted-filter-open-portal-id"
        />
      </div>
      <div
        id="assorted-filter-close-portal-id"
      />
    </div>
  </div>
  <div
    className="resourceDiagramControlsContainer"
  >
    <div
      className="diagram-title"
    >
      Resource topology
    </div>
    <div
      className="topologyDiagramContainer"
    >
      <div
        className="diagramViewerDiagram"
      >
        <div
          className="diagramViewerContainerContainer"
          id="diagram-viewer-container-container"
        >
          <div
            aria-label="zoom"
            className="diagramViewerContainer"
            role="region"
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <svg
              className="topologyDiagram"
              id="topologySvgId"
            />
          </div>
          <div
            className="secondaryLoad"
          >
            <div
              aria-live="assertive"
              className="bx--loading"
            >
              <svg
                className="bx--loading__svg"
                viewBox="-75 -75 150 150"
              >
                <circle
                  cx="0"
                  cy="0"
                  r="37.5"
                />
              </svg>
            </div>
          </div>
          <div
            className="layoutLoadingContainer"
          >
            <div
              aria-live="assertive"
              className="bx--loading"
            >
              <svg
                className="bx--loading__svg"
                viewBox="-75 -75 150 150"
              >
                <circle
                  cx="0"
                  cy="0"
                  r="37.5"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="diagram-controls"
        >
          <div
            aria-label="Zoom in"
            className="zoom-in-button rectangle-zoom"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            title="Zoom in"
          >
            <svg
              className="icon"
            >
              <use
                href="#diagramIcons_zoomIn"
              />
            </svg>
          </div>
          <div
            aria-label="Zoom out"
            className="zoom-out-button rectangle-zoom"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            title="Zoom out"
          >
            <svg
              className="icon"
            >
              <use
                href="#diagramIcons_zoomOut"
              />
            </svg>
          </div>
          <div
            aria-label="Zoom to fit"
            className="zoom-target-button rectangle-zoom"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            title="Zoom to fit"
          >
            <svg
              className="icon_target"
            >
              <use
                href="#diagramIcons_zoomTarget"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="diagram-controls-container"
    >
      <div
        className="diagram-type-filter-bar"
        id="type-filter-bar-portal-id"
      />
      <div
        aria-label="View topology"
        className="diagram-expand-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
        title="View topology"
      >
        <svg
          className="icon"
        >
          <use
            href="#diagramIcons_launch"
          />
        </svg>
        View topology
      </div>
    </div>
  </div>
</div>
`;
